torchgeo/datasets/spacenet1.py

```diff
@@ -0,0 +1,215 @@
+                )
+
+        super().__init__(
+            root=os.path.join(root, self.foldername),
+            crs=crs,
+            res=res,
+            transforms=transforms,
+        )
+        self.files = self._load_files(os.path.join(root, self.foldername))
+
+    def _load_files(self, root) -> List[Dict[str, str]]:
+        files = []
+        images = glob.glob(os.path.join(root, "*", self.filename_glob))
+        images = sorted(images)
+        for imgpath in images:
+            rawpath = imgpath.replace("RGB.tif", "8Band.tif")
+            lbl_path = os.path.join(
+                os.path.dirname(rawpath) + "-labels", "labels.geojson"
+            )
+            files.append({"rgb": imgpath, "8band": rawpath, "label": lbl_path})
+        return files
+
+    def _load_image(self, path: str) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            the image
+        """
+        filename = os.path.join(path)
+        with rio.open(filename) as img:
+            array = img.read().astype(np.float32)
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            return tensor, img.transform
+
+    def _load_label(self, path: str, tfm: Affine, shape: Tuple[int, int]) -> Tensor:
+        """Rasterizes the dataset's labels (in geojson format).
+
+        Args:
+            path (str): path to the label
+            tfm (Affine): transform of corresponding image
+            shape (List[int, int]): shape of corresponding image
+
+        Returns:
+            Tensor: label tensor
+        """
+
+        with fiona.open(path) as src:
+            labels = [feature["geometry"] for feature in src]
+
+        if not labels:
+            mask_data = np.zeros(shape=shape)
+        else:
+            mask_data = rasterize(
+                labels,
+                out_shape=shape,
+                fill=0,  # nodata value
+                transform=tfm,
+                all_touched=False,
+                dtype=np.uint8,
+            )
+
+        mask: Tensor = torch.from_numpy(mask_data).long()  # type: ignore[attr-defined]
+
+        return mask
+
+    def __len__(self) -> int:
+        """Return the number of samples in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.files)
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        files = self.files[index]
+        rgb, tfm = self._load_image(files["rgb"])
+        raw, _ = self._load_image(files["8band"])
+        out_shape = tuple(rgb.shape[1:])
+        label = self._load_label(files["label"], tfm, out_shape)
+
+        sample = {"rgb": rgb, "8band": raw, "label": label}
```

Comment:
We should figure out a better way to handle this. If every dataset uses custom dictionary keys, there's no way for a transform to operate on a sample and know which key to use. For example, if I want to do a random horizontal flip, I need to know all keys that need to be flipped ahead of time. We usually stack all images in a single 'image' key for this reason.

Reply:
Possibly need to define as an input arg to the constructor whether the output "image" should be the RGB or 8 band image. Is there a reason why someone would need both for training/inference?

Reply:
I believe both were given to allow challenge participants to experiment on different imagery. This is also the case for other spacenet datasets (2,4,5) where they provided PAN (panchromatic), MS (multi-spectral) and PS-MS (pan-sharpened multi-spectral) imagery.

Reply:
Sounds good, I like @isaaccorley's suggestion of an argument that decides which type of imagery is returned. We can think about how to return both at the same time in a future revision.

Reply:
Added `image` arg to allow users to choose between RGB and 8 band imagery.